src/index.ts

```diff
@@ -174,9 +176,12 @@ const run = async () => {
     );
 
     for (const pullRequest of pullRequests) {
-      // PRs are handled sequentially to avoid breaking GitHub's log grouping feature.
-      // eslint-disable-next-line no-await-in-loop
-      await handlePullRequest(pullRequest, { eventPayload, octokit });
+      if (pullRequest.auto_merge) {
+        // PRs are handled sequentially to avoid breaking GitHub's log grouping feature.
+        // eslint-disable-next-line no-await-in-loop
+        await handlePullRequest(pullRequest, { eventPayload, octokit });
+        break;
```

Comment:
I would lean into an early `return` here rather than a `break` if there's nothing for us to do once we're out, anyway.

Reply:
great point, will do
```suggestion
        return;
```